File: DictHashMaps/twoStrings.py
# Complete the twoStrings function below.
def twoStrings(s1, s2):
    for letter in s1:
        if letter in s2:
            return 'YES'
    return 'NO'


if __name__ == '__main__':
    s1 = 'hi'
    s2 = 'world'
    result = twoStrings(s1, s2)
    print(result)

File: DictHashMaps/twoStrings.js
function maximumToys(s1, s2) {
    var map = {}
    for (let letter of s1) {
        map[letter] = 1;
    }
    
    for (let letter of s2) {
        if (map[letter]) {
            return 'YES'
        }
    }
    return 'NO'
}


// testings
const testStrings = [['hello', 'llot'], ['1', '2']]
for (i in testStrings) {
    const result = maximumToys(...testStrings[i])
    console.log(result)
}
